src/interactivity/win32/window.cpp

```diff
@@ -68,10 +67,13 @@ Window::Window() :
 
 Window::~Window()
 {
-    if (ServiceLocator::LocateGlobals().pRender != nullptr)
-    {
-        delete ServiceLocator::LocateGlobals().pRender;
-    }
+    delete pGdiEngine;
+#if TIL_FEATURE_CONHOSTDXENGINE_ENABLED
+    delete pDxEngine;
+#endif
+#if TIL_FEATURE_ATLASENGINE_ENABLED
+    delete pAtlasEngine;
+#endif

```

Comment:
Wait, the Renderer doesn't `delete` its engines on teardown...?

Reply:
No, it only has weak references (via standard pointers). The `ControlCore` owns it.